### DISCUSSION TITLE
Signature generálás PHP-ban

### DISCUSSION THREAD

**Discussion by NoviusIT** [nav-gov-hu/M2M#39] [Kérdések]
Üdv.
Egy olyan gondom lenne, hogy éles környezetben próbálok felhasználó aktiváláshoz aláírást generálni, de nem jön össze.
Mi lehet a probléma?
```
$to_sign = $message_id . $timestamp . $signature_key_first_part . $signature_key_second_part;
$signature = strtoupper(base64_encode(hash('sha256', $to_sign)));
```

A message_id egy generált azonosító a dokumentációban megadott pattern alapján ```[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}```
A timestamp egy date("YmdHis"), illetve nonce-al sikerült lekérnem az aláíró kód 2. felét.

Köszönöm a segítséget.

**Comment by czeba-b** [comment]
Szia!

Nekem egyelőre a tesztrendszeren van működő aláírás generálásom python-ban, azzal összehasonlítva a kódod nem ugyanazt a hash-t adja ugyanarra a `to_sign` stringre.

to_sign: `00000000-0000-0000-0000-0000000000002024112208160111111111112222222222`
signature nálam: `HAABVXD1SO6SNKSBEMEQNWPSRXRU3GJBXNJDYJMRTSC=`
signature nálad: `MWRHNJGXNTU3MGY1NDHLZWFJOWNHNDLINZHJMTEWOWQ2M2QYNDCXYWQ0ZGUWMJVIYZRKMJVKNJIZOWFIYJZJNW==`

Viszont ha a kódodban átírom a hash hívást, hogy binárist adjon vissza akkor már ugyanazt adja: 
```php
strtoupper(base64_encode(hash('sha256', $to_sign, true)))
```

**Reply by NoviusIT** [reply]
Szia!

Köszönöm, de sajnos így is INVALID_SIGNATURE -t kapok, illetve így nem felel meg a pattern-nek `[0-9a-fA-F]{44,64}` , nem tudom mi lehet a gond. Ha beleteszek még egy bin2hex() -et, akkor a hossznak nem felel meg, ellenben a hossznak megfelel, minden másnak nem.
